okhttp/src/main/java/io/grpc/transport/okhttp/OkHttpClientTransport.java

```diff
@@ -576,7 +642,12 @@ public void ackSettings() {
 
     @Override
     public void goAway(int lastGoodStreamId, ErrorCode errorCode, ByteString debugData) {
-      onGoAway(lastGoodStreamId, Status.UNAVAILABLE.withDescription("Go away"));
+      Status status = HttpUtil.Http2Error.statusForCode(errorCode.httpCode);
```

Comment:
This is also for consistency with the Netty implementation (and makes more sense to properly map the `errorCode` to a corresponding `Status.Code`).